pwiz_tools/Skyline/SettingsUI/FormulaBox.cs

```diff
@@ -420,6 +422,15 @@ private void AddFormulaSymbol(string symbol)
             textFormula.SelectionStart = insertAt + symbol.Length;
         }
 
+        private void SetFormulaText(string text)
+        {
+            // Preserve cursor location

```

Comment:
I am not sure if it would make a difference here, but you might want to add:
`if (text == textFormula.Text) return;`
so that nothing happens if the text is not actually changing.

Reply:
Certainly can't hurt.